From 4cbaa4be3dff1c1c20db87e37f15c64ff2c7befc Mon Sep 17 00:00:00 2001
From: Matt Ranostay <matt.ranostay@intel.com>
Date: Wed, 11 Feb 2015 11:38:13 -0800
Subject: [PATCH] bmc150: move IIO_EV_INFO* flags to shared mask

Enable and period are shared mask attributes, and not configurable
as separate X, Y, Z channel settings.

Change-Id: I0e7c9069817ae25d70dd4da313acadd71165639a
Signed-off-by: Matt Ranostay <matt.ranostay@intel.com>
---
 drivers/iio/accel/bmc150-accel.c | 6 +++---
 1 file changed, 3 insertions(+), 3 deletions(-)

diff --git a/drivers/iio/accel/bmc150-accel.c b/drivers/iio/accel/bmc150-accel.c
index 3eae194..2d280ac 100644
--- a/drivers/iio/accel/bmc150-accel.c
+++ b/drivers/iio/accel/bmc150-accel.c
@@ -873,9 +873,9 @@ static const struct attribute_group bmc150_accel_attrs_group = {
 static const struct iio_event_spec bmc150_accel_event = {
 		.type = IIO_EV_TYPE_ROC,
 		.dir = IIO_EV_DIR_EITHER,
-		.mask_separate = BIT(IIO_EV_INFO_VALUE) |
-				 BIT(IIO_EV_INFO_ENABLE) |
-				 BIT(IIO_EV_INFO_PERIOD)
+		.mask_separate = BIT(IIO_EV_INFO_VALUE),
+		.mask_shared_by_type = BIT(IIO_EV_INFO_ENABLE) |
+				BIT(IIO_EV_INFO_PERIOD)
 };
 
 #define BMC150_ACCEL_CHANNEL(_axis, bits) {				\
-- 
1.9.1

